.github/workflows/build-and-push.yml

```diff
@@ -2,40 +2,74 @@ name: Build and Push
 
 on:
   push:
-    branches: [main]
-    tags:
-      - v*
+    branches: [main, "release-[0-9].[0-9]+"]
+    paths-ignore: ["**.md"]
+    tags: ["v*"]
+  pull_request:
+    branches: [main, "release-[0-9].[0-9]+"]
+    paths-ignore: ["**.md"]
+
+env:
+  IMAGE_NAME: "kserve/modelmesh-runtime-adapter"
 
 jobs:
-  test:
```

Comment:
Yes, but unit tests are run in a separate workflow already anyhow, and would fail independently, so this reduces time and resource consumption on the GitHub workflow runner. Once a PR is merged, this `build-and-push` workflow will run without test since there is no good way to inform/alert anyone of a failure and via branch protection rules we can assume that only PRs where tests succeed get merged.

Reply:
additionally, the `push:` key currently evaluates to false on PRs so the registry isn't polluted.

fwiw a pattern I regularly see is a simple RUN statement at the end of the Dockerfile that functions as a smoketest for CI, e.g. the entrypoint is set up properly and has access to whatever was COPY'd. if this RUN statement doesn't write files to disk, it doesn't add to the final image size.

Reply:
Thanks for the context and details. Yeah, It seems unlikely that this would actually cause an image to be pushed that doesn't pass tests so long as no one pushes directly to `main`.

Reply:
I disabled the ability to push directly to main, even for project owners/maintainers